lls/main.c

```diff
@@ -719,6 +806,63 @@ register_nd_acl_rules(struct gatekeeper_if *iface)
 	return 0;
 }
 
+static void
+fill_ping6_rule(struct ipv6_acl_rule *rule, struct in6_addr *addr)
+{
+	uint32_t *ptr32 = (uint32_t *)addr;
+	int i;
+
+	rule->data.category_mask = 0x1;
+	rule->data.priority = 1;
+	/* Userdata is filled in in register_ipv6_acl(). */
+
+	rule->field[PROTO_FIELD_IPV6].value.u8 = IPPROTO_ICMPV6;
+	rule->field[PROTO_FIELD_IPV6].mask_range.u8 = 0xFF;
+
+	for (i = DST1_FIELD_IPV6; i <= DST4_FIELD_IPV6; i++) {
+		rule->field[i].value.u32 = rte_be_to_cpu_32(*ptr32);
+		rule->field[i].mask_range.u32 = 32;
+		ptr32++;
+	}
+
+	/*
+	 * The reason to have 0xFFFF0000 is to test
+	 * both type and code at the same time.
+	 */
+	rule->field[TYPE_FIELD_ICMPV6].value.u32 =
+		((ICMPV6_ECHO_REQUEST_TYPE << 24) +
+		(ICMPV6_ECHO_REQUEST_CODE << 16)) & 0xFFFF0000;
```

Comment:
Style: replace the `+` with `|`.

This comment also applies to patch **lls: respond to ICMP pings**